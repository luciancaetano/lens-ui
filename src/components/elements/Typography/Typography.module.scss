@import '../../../style-lib';
@import '../../../typography-lib';

.root {
  @include typography-base;
}

.h1 {
  @include h1;
}

.h2 {
  @include h2;
}

.h3 {
  @include h3;
}

.h4 {
  @include h4;
}

.h5 {
  @include h5;
}

.h6 {
  @include h6;
}

.subtitle1 {
  @include subtitle1;
}

.subtitle2 {
  @include subtitle2;
}

.body1 {
  @include body1;
}

.body2 {
  @include body2;
}

.caption {
  @include caption;
}

.overline {
  @include overline;
}

.button {
  @include button;
}

.no-wrap {
  @include no-wrap;
}
