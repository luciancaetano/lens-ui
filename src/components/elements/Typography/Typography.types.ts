import React from 'react';
import { IntentType, ITestableProps } from '../../../types';

export interface ITypographyProps extends ITestableProps, Omit<React.HTMLAttributes<HTMLElement>, 'onChange'> {
  variant: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'subtitle1' | 'subtitle2' |'body1' | 'body2' | 'overline' | 'button' |'caption' ;
  align?: 'inherit' | 'left' | 'center' | 'right' | 'justify';
  noWrap?: boolean;
  children: React.ReactNode;
  intent?: IntentType | null;
  gutterBottom?: boolean;
}
