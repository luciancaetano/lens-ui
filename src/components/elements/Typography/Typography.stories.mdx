import { Canvas, Meta, Story } from '@storybook/addon-docs';
import Typography from './Typography';


<Meta title="Components/Typography" component={Typography}/>

# Typography

Typography is a component that renders text with a specific font.

<Canvas>
  <div style={{ maxWidth: 700 }}>
    <Typography variant="h1">h1. Heading</Typography>
    <Typography variant="h2">h2. Heading</Typography>
    <Typography variant="h3">h3. Heading</Typography>
    <Typography variant="h4">h4. Heading</Typography>
    <Typography variant="h5">h5. Heading</Typography>
    <Typography variant="h6">h6. Heading</Typography>
    <Typography variant="subtitle1">subtitle1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur</Typography>
    <Typography variant="subtitle2">subtitle2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur</Typography>
    <Typography variant="body1">
      body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit,
      quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
    </Typography>
    <Typography variant="body2">
      body2. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus,
      cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
    </Typography>
    <Typography variant="button">BUTTON TEXT</Typography>
    <Typography variant="caption">caption text</Typography>
    <Typography variant="overline">OVERLINE TEXT</Typography>
  </div>
</Canvas>

### Properties

|Property|Type|Default Value|Description|Required|
|---|---|---|---|---|
|variant|string|'body1'|The variant of the typography.|Yes|
|align|string|'inherit'|The alignment of the typography.|No|
|noWrap|boolean|false|Whether or not the typography should wrap.|No|
|intent|string|null|The intent of the typography.|No|
|gutterBottom|boolean|false | Whether to add a bottom margin.|No|
|testingID| string | - | The id used for testing purposes.<br/>`<div data-testid="my-test-id"/>` |No|
