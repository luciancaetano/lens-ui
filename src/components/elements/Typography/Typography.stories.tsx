/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { ComponentMeta } from '@storybook/react';
import Typography from './Typography';
import LensProvider from '../../providers/LensProvider/LensProvider';
import '../../../styles';

export default {
  title: '2. Components/Typography',
  component: Typography,
  decorators: [
    (Story) => <LensProvider>{Story()}</LensProvider>,
  ],
} as ComponentMeta<typeof Typography>;

export const Default = () => (
  <div style={{ maxWidth: 700 }}>
    <Typography variant="h1">h1. Heading</Typography>
    <Typography variant="h2">h2. Heading</Typography>
    <Typography variant="h3">h3. Heading</Typography>
    <Typography variant="h4">h4. Heading</Typography>
    <Typography variant="h5">h5. Heading</Typography>
    <Typography variant="h6">h6. Heading</Typography>
    <Typography variant="subtitle1">subtitle1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur</Typography>
    <Typography variant="subtitle2">subtitle2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur</Typography>
    <Typography variant="body1">
      body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit,
      quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
    </Typography>
    <Typography variant="body2">
      body2. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus,
      cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
    </Typography>
    <Typography variant="button">BUTTON TEXT</Typography>
    <Typography variant="caption">caption text</Typography>
    <Typography variant="overline">OVERLINE TEXT</Typography>
  </div>
);
