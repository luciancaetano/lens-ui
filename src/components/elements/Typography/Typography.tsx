import React from 'react';
import clsx from 'clsx';
import { ITypographyProps } from './Typography.types';
import styles from './Typography.module.scss';

const variantMap: { [key: string]: string } = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
  h5: 'h5',
  h6: 'h6',
  subtitle1: 'h6',
  subtitle2: 'h6',
  body1: 'p',
  body2: 'p',
  caption: 'span',
  overline: 'span',
};

const Typography: React.FC<ITypographyProps> = ({ variant, noWrap }) => React.createElement(variantMap[variant], {
  style: {
    whiteSpace: noWrap ? 'nowrap' : 'normal',
  },
  className: clsx(styles.root, styles[variant]),
}, 'Typography');

export default Typography;
