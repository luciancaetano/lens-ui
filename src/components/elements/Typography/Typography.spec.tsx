import React from 'react';
import { render } from '@testing-library/react';
import Typography from './Typography';

describe('<Typography/>', () => {
  it('should render', () => {
    const { container } = render(
      <Typography variant="h1">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with noWrap', () => {
    const { container } = render(
      <Typography variant="h1" noWrap>
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with intent', () => {
    const { container } = render(
      <Typography variant="h1" intent="primary">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with children', () => {
    const { container } = render(
      <Typography variant="h1">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with children and noWrap', () => {
    const { container } = render(
      <Typography variant="h1" noWrap>
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with children and intent', () => {
    const { container } = render(
      <Typography variant="h1" intent="primary">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with children and noWrap and intent', () => {
    const { container } = render(
      <Typography variant="h1" noWrap intent="primary">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  // test variants

  it('should render with variant h1', () => {
    const { container } = render(
      <Typography variant="h1">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant h2', () => {
    const { container } = render(
      <Typography variant="h2">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant h3', () => {
    const { container } = render(
      <Typography variant="h3">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant h4', () => {
    const { container } = render(
      <Typography variant="h4">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant h5', () => {
    const { container } = render(
      <Typography variant="h5">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant h6', () => {
    const { container } = render(
      <Typography variant="h6">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant subtitle1', () => {
    const { container } = render(
      <Typography variant="subtitle1">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant subtitle2', () => {
    const { container } = render(
      <Typography variant="subtitle2">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant body1', () => {
    const { container } = render(
      <Typography variant="body1">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant body2', () => {
    const { container } = render(
      <Typography variant="body2">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant caption', () => {
    const { container } = render(
      <Typography variant="caption">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant button', () => {
    const { container } = render(
      <Typography variant="button">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with variant overline', () => {
    const { container } = render(
      <Typography variant="overline">
        Hello World
      </Typography>,
    );
    expect(container).toMatchSnapshot();
  });
});
