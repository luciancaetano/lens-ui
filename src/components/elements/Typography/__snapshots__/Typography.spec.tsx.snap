// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography/> should render 1`] = `
<div>
  <h1
    class="root h1"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with children 1`] = `
<div>
  <h1
    class="root h1"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with children and intent 1`] = `
<div>
  <h1
    class="root h1"
    data-lens-element="typography"
    style="--typography-intent-color: var(--lens-ui-intents-primary); --typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with children and noWrap 1`] = `
<div>
  <h1
    class="root h1 noWrap"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with children and noWrap and intent 1`] = `
<div>
  <h1
    class="root h1 noWrap"
    data-lens-element="typography"
    style="--typography-intent-color: var(--lens-ui-intents-primary); --typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with intent 1`] = `
<div>
  <h1
    class="root h1"
    data-lens-element="typography"
    style="--typography-intent-color: var(--lens-ui-intents-primary); --typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with noWrap 1`] = `
<div>
  <h1
    class="root h1 noWrap"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with variant body1 1`] = `
<div>
  <p
    class="root body1"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </p>
</div>
`;

exports[`<Typography/> should render with variant body2 1`] = `
<div>
  <p
    class="root body2"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </p>
</div>
`;

exports[`<Typography/> should render with variant button 1`] = `
<div>
  <span
    class="root button"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </span>
</div>
`;

exports[`<Typography/> should render with variant caption 1`] = `
<div>
  <span
    class="root caption"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </span>
</div>
`;

exports[`<Typography/> should render with variant h1 1`] = `
<div>
  <h1
    class="root h1"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h1>
</div>
`;

exports[`<Typography/> should render with variant h2 1`] = `
<div>
  <h2
    class="root h2"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h2>
</div>
`;

exports[`<Typography/> should render with variant h3 1`] = `
<div>
  <h3
    class="root h3"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h3>
</div>
`;

exports[`<Typography/> should render with variant h4 1`] = `
<div>
  <h4
    class="root h4"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h4>
</div>
`;

exports[`<Typography/> should render with variant h5 1`] = `
<div>
  <h5
    class="root h5"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h5>
</div>
`;

exports[`<Typography/> should render with variant h6 1`] = `
<div>
  <h6
    class="root h6"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h6>
</div>
`;

exports[`<Typography/> should render with variant overline 1`] = `
<div>
  <span
    class="root overline"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </span>
</div>
`;

exports[`<Typography/> should render with variant subtitle1 1`] = `
<div>
  <h6
    class="root subtitle1"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h6>
</div>
`;

exports[`<Typography/> should render with variant subtitle2 1`] = `
<div>
  <h6
    class="root subtitle2"
    data-lens-element="typography"
    style="--typography-align: inherit; --typography-gutter-bottom: 0;"
  >
    Hello World
  </h6>
</div>
`;
